roboteq_diff_driver/src/driver.cpp

```diff
@@ -655,17 +236,39 @@ void MainNode::odom_loop()
           if (odom_buf[delim] == ':')
           {
             odom_buf[delim] = 0;
-            odom_encoder_right = (int32_t)strtol(odom_buf+3, NULL, 10);
-            odom_encoder_left = (int32_t)strtol(odom_buf+delim+1, NULL, 10);
-#ifdef _ODOM_DEBUG
-ROS_DEBUG_STREAM("encoder right: " << odom_encoder_right << " left: " << odom_encoder_left);
-#endif
+            /* the query in line 200 asks for information every 25ms, hence the 0.025
+            * the factor 4 is a scaling factor of unknown origin, probably has to do something with cpr-ppr
+            * values are very precise (verified against baseline), so 4 is probably the exact value */
+            rpm_right_meas = -(int32_t)strtol(odom_buf+3, NULL, 10)*60./(0.025*4*encoder_cpr); 
+            rpm_left_meas = -(int32_t)strtol(odom_buf+delim+1, NULL, 10)*60./(0.025*4*encoder_cpr);
+            if (debug_){
+            ROS_DEBUG_STREAM("rpm encoder right: " << odom_encoder_right << " left: " << odom_encoder_left);
+            }
             odom_publish();
             break;
           }
         }
-      }
-#ifdef _ODOM_SENSORS
+      } 
+      if ( odom_buf[0] == 'T' && odom_buf[1] == 'C' && odom_buf[2] == '=') //TC is current torque command
+      {
+        int delim;
+        for ( delim = 3; delim < odom_idx; delim++ )
+        {
+          if ( odom_encoder_toss > 0 )
+          {
+            --odom_encoder_toss;
+            break;
+          }
+          if (odom_buf[delim] == ':')
+          {
+            odom_buf[delim] = 0;
+            torque_right_meas = (int32_t)strtol(odom_buf+3, NULL, 10)/100.0;
```

Comment:
Also explain the factor of 100 (It's the resolution it's told to send data at right?)

Reply:
Correct, added